contrib/repairStatus.py

```diff
@@ -0,0 +1,199 @@
+                                    "to": current_time
+                                }
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'host:("' + node + '")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'method:("syncComplete")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'message:("is fully synced")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    }
+  }
+
+  return json.dumps(query_json)
+
+
+# Execute an ES _count to return the number of ranges that have finished repairing
+def calculate_range_repairs(host, port, node, start_time, current_time):
+  url = 'http://%(host)s:%(port)s/_count' % dict(host=host, port=port)
+  query = generate_range_repair_query(node, start_time, current_time)
+  
+  log.info("Ranges completed: curl %(url)s -d '%(query)s'" % dict(url=url, query=query)) 
+  r = requests.get(url, data=query)
+  
+  if r.status_code == 200:
+    response = json.loads(r.text)
+    return response['count']
+
+
+def generate_total_ranges_query(node, start_time, current_time):
+  query_json = {
+    "fields": "message",
+    "query": {
+      "filtered": {
+        "filter": {
+          "bool": {
+            "must": [
+              {
+                "range": {
+                  "@timestamp": {
+                    "to": current_time,
+                    "from": start_time
+                  }
+                }
+              },
+              {
+                "fquery": {
+                  "_cache": True,
+                  "query": {
+                    "query_string": {
+                      "query": 'host:("' + node + '")'
+                    }
+                  }
+                }
+              }
+            ]
+          }
+        },
+        "query": {
+          "query_string": {
+            "query": 'message: "Starting repair"'
+          }
+        }
+      }
+    }
+  }
```

Comment:
This is more of a personal taste thing, nothing to do with Python style, but I found it much easier to read to compress this into fewer lines. Emacs helped a lot with this, but what I came up with was:
```python
    query_json = {
        "fields": "message",
        "query":
        {"filtered":
         {"filter":
          {"bool": {"must": [
              {"range":
               {"@timestamp": {"from": start_time, "to": current_time}}},
              {"fquery": {"_cache": True,
                          "query": {"query_string":
                                    {"query": 'host:("' + node + '")'}}}}
          ]}},
      "query": {"query_string": {"query": 'message: "Starting repair"'}}}}
    }
```
This one you can really take or leave :).